modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/LayoutReportsApp.js

```diff
@@ -12,33 +12,211 @@
  * details.
  */
 
-import React from 'react';
+import ClayAlert from '@clayui/alert';
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayLayout from '@clayui/layout';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import {useEventListener, useIsMounted} from '@liferay/frontend-js-react-web';
+import {fetch} from 'frontend-js-web';
+import React, {
+	useCallback,
+	useEffect,
+	useLayoutEffect,
+	useReducer,
+	useState,
+} from 'react';
 
 import BasicInformation from './components/BasicInformation';
 import EmptyLayoutReports from './components/EmptyLayoutReports';
 
-export default function ({context}) {
-	const {
-		assetsPath,
-		canonicalURLs,
-		configureGooglePageSpeedURL,
-		defaultLanguageId,
-		validConnection,
-	} = context;
+const initialState = {
+	data: null,
+	error: null,
+	loading: false,
+};
+
+const dataReducer = (state, action) => {
+	switch (action.type) {
+		case 'LOAD_DATA':
+			return {
+				...state,
+				loading: true,
+			};
+
+		case 'SET_ERROR':
+			return {
+				...state,
+				error: action.error,
+				loading: false,
+			};
+
+		case 'SET_DATA':
+			return {
+				data: {
+					...action.data,
+				},
+				error: action.data?.error,
+				loading: false,
+			};
+
+		default:
+			return initialState;
+	}
+};
+
+export default function ({
+	isPanelStateOpen,
+	layoutReportsDataURL,
+	portletNamespace,
+}) {
+	const isMounted = useIsMounted();
+
+	const [state, dispatch] = useReducer(dataReducer, initialState);
+
+	const safeDispatch = (action) => {
+		if (isMounted()) {
+			dispatch(action);
+		}
+	};
+
+	const [isLoaded, setIsLoaded] = useState(false);
```

Comment:
This state is initialize in every render, so `getData` is called again when `mouseenter` or `focus` happen.